frame/ethereum/src/lib.rs

```diff
@@ -207,11 +215,22 @@ decl_module! {
 		}
 
 		fn on_finalize(n: T::BlockNumber) {
-			<Module<T>>::store_block();
+			<Module<T>>::store_block(
+				fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_err(),
+			);
 		}
 
 		fn on_initialize(n: T::BlockNumber) -> Weight {
 			Pending::kill();
+
+			if let Ok(log) = fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()) {
+				let PreLog::Block(block) = log;
+
+				for transaction in block.transactions {
+					let _ = Call::<T>::transact(transaction).dispatch_bypass_filter(RawOrigin::None.into());
+				}
```

Comment:
It looks like this will always fail, because `fp_consensus::find_pre_log(&frame_system::Module::<T>::digest()).is_err()
` is false, see Line 137.